herddb-core/src/main/java/herddb/model/Table.java

```diff
@@ -176,13 +179,17 @@ public static Table deserialize(byte[] data) {
             for (int i = 0; i < ncols; i++) {
                 long cversion = dii.readVLong(); // version
                 long cflags = dii.readVLong(); // flags for future implementations
-                if (cversion != 1 || cflags != 0) {
+                if (cversion != 1 || (cflags != 0 && cflags != COLUMNFLAGS_HAS_DEFAULT_VALUE)) {
```

Comment:
Couldn't just be (cversion != 1 || cflags != COLUMNFLAGS_HAS_DEFAULT_VALUE)?

Reply:
actually allowed values are:
0 -> NO FLAG
1 -> COLUMNFLAGS_HAS_DEFAULT_VALUE

we still have to allow cflags == 0 in order to be compatible  with old versions.
we are now still keeping flags = 0 if there is no default value, in order to allow rollbacks to previous versions in case you did not use the new DEFAULT VALUE feature

btw I have introduced constants